data_steward/validation/main.py

```diff
@@ -52,49 +52,65 @@ def all_required_files_loaded(hpo_id, folder_prefix):
     return True
 
 
-def save_datasources_json(hpo_id, folder_prefix=""):
-    hpo_bucket = gcs_utils.get_hpo_bucket(hpo_id)
+def save_datasources_json(hpo_id=None, folder_prefix="", target_bucket=None):
+    if hpo_id is None and target_bucket is None:
```

Comment:
As discussed, if all three args are supplied, report should be in `target_bucket` and use `hpo_id` as the name of the data source.